core/src/main/java/org/elasticsearch/index/percolator/PercolatorQueriesRegistry.java

```diff
@@ -208,12 +208,12 @@ private Query parseQuery(String type, XContentParser parser) {
             context.setMapUnmappedFieldAsString(mapUnmappedFieldsAsString ? true : false);
             return queryParserService.parseInnerQuery(context);
         } catch (IOException e) {
-            throw new QueryParsingException(context, "Failed to parse", e);
+            throw new QueryCreationException(context, "Failed to parse", e);
```

Comment:
seems weird to have a query creation exception with text "failed to parse". This is still a method that parse all the way from xcontent to lucene query, not 100% sure what we should be doing here. Maybe the change is ok but we need to add some TODO or //norelease that says we have to get back to this method and split it in two phases at some point.